crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -640,39 +698,106 @@ pub mod tests {
 
         let binding_address: SocketAddr = "0.0.0.0:9185".parse().unwrap();
         let registry = mysten_metrics::start_prometheus_server(binding_address).default_registry();
-        let metrics = RequestMetrics::new(&registry);
-        let metrics = Arc::new(metrics);
-        let metrics2 = metrics.clone();
-
+        let metrics = Metrics::new(&registry);
+        let _metrics2 = metrics.request_metrics.clone();
         let service_config = ServiceConfig::default();
         let db_url: String = connection_config.db_url.clone();
         let reader = PgManager::reader(db_url).expect("Failed to create pg connection pool");
-        let db = Db::new(reader.clone(), service_config.limits);
+        let db = Db::new(reader.clone(), service_config.limits, metrics.clone());
         let pg_conn_pool = PgManager::new(reader, service_config.limits);
-        let schema = ServerBuilder::new(8000, "127.0.0.1".to_string())
+        let schema = ServerBuilder::new(8000, "127.0.0.1".to_string(), metrics.clone())
             .context_data(db)
             .context_data(pg_conn_pool)
             .context_data(service_config)
-            .context_data(metrics)
+            .context_data(metrics.clone())
+            .context_data(query_id())
+            .context_data(ip_address())
             .extension(QueryLimitsChecker::default())
             .build_schema();
         let _ = schema.execute("{ chainIdentifier }").await;
 
-        assert_eq!(metrics2.input_nodes.get_sample_count(), 1);
-        assert_eq!(metrics2.output_nodes.get_sample_count(), 1);
-        assert_eq!(metrics2.query_depth.get_sample_count(), 1);
-        assert_eq!(metrics2.input_nodes.get_sample_sum(), 1.);
-        assert_eq!(metrics2.output_nodes.get_sample_sum(), 1.);
-        assert_eq!(metrics2.query_depth.get_sample_sum(), 1.);
+        // TODO these do not get properly triggered. Need more investigation, seems to be
```

Comment:
Flagging this part for clean-up -- but I'm not sure what the right thing to do is here -- will need someone who is familiar with `mysten-metrics` to have a look.

Reply:
I scheduled some time with @suiwombat to look together at this, and will ping other folks as needed.